cmake/setup_cached_variables.cmake

```diff
@@ -129,6 +129,17 @@ IF( NOT "${CMAKE_BUILD_TYPE}" STREQUAL "Release" AND
 ENDIF()
 
 #
+# Force exactly one build type for multi-cofiguration generators.
+# Unfortunately CMake doesn't support custom configuration types so we just
+# call "DebugRelease" a "Release" build as well.
+#
+IF(CMAKE_BUILD_TYPE MATCHES "Release")
+  SET(CMAKE_CONFIGURATION_TYPES "Release" CACHE STRING "" FORCE)
+ELSE()
+  SET(CMAKE_CONFIGURATION_TYPES "Debug" CACHE STRING "" FORCE)
```

Comment:
Unfortunately, the macro ```DEAL_II_SETUP_TARGET``` is currently also not multi-configuration aware. This was a semiconscious decision to support some features @luca-heltai asked to have :-)

So bottom line is that I would like to
 - disable the whole multi-configuration business for the library itself (which is this PR)
 - fix the testsuite subprojects (in a follow up PR) and
 - discuss whether we want to support multi-configuration setups on user side (which will require to fix the ```DEAL_II_SETUP_TARGET``` macro.